src/mono/System.Private.CoreLib/src/System/Array.Mono.cs

```diff
@@ -55,6 +55,20 @@ public int Rank
             get => Rank;
         }
 
+        internal static unsafe void Clear(Array array)
```

Comment:
Why not just call AsSpan().Clear() as needed? It would be typically smaller and faster code.

Reply:
Ah, I guess that would run into issues with shared generic code inlining.

Reply:
Some is the shared code issue, some is that I don't want to risk array variance checks causing the `AsSpan()` call to fail in scenarios where it otherwise would've worked. (Dictionary and friends don't suffer from the variance problem.)